install.php

```diff
@@ -207,6 +250,186 @@ public function installOrUpdate($parent)
+						if (substr(get_class(JFactory::$database), 0, 1) == 'J' && $this->extensionName != 'com_redcore')
+						{
+							RFactory::$database = null;
+							JFactory::$database = null;
+							JFactory::$database = RFactory::getDbo();
+						}
+
+						$db = JFactory::getDbo();
+
+						$dbDriver = strtolower($db->name);
+						$schemapath = '';
+
+						if ($dbDriver == 'mysqli')
+						{
+							$dbDriver = 'mysql';
+						}
+
+						foreach ($schemapaths as $entry)
+						{
+							if (isset($entry->attributes()->type))
+							{
+								$uDriver = strtolower($entry->attributes()->type);
+
+								if ($uDriver == 'mysqli')
+								{
+									$uDriver = 'mysql';
+								}
+
+								if ($uDriver == $dbDriver)
+								{
+									$schemapath = (string) $entry;
+									break;
+								}
+							}
+						}
+
+						if ($schemapath != '')
+						{
+							$files = str_replace('.sql', '', JFolder::files($sourcePath . '/' . $path . '/' . $schemapath, '\.sql$'));
+							usort($files, 'version_compare');
+
+							if (!count($files))
+							{
+								return false;
+							}
+
+							foreach ($files as $file)
+							{
+								if (version_compare($file, $this->oldVersion) > 0)
+								{
+									$buffer = file_get_contents($sourcePath . '/' . $path . '/' . $schemapath . '/' . $file . '.sql');
+									$queries = RDatabaseDriver::splitSQL($buffer);
+
+									if (count($queries))
+									{
+										foreach ($queries as $query)
+										{
+											if ($query != '' && $query{0} != '#')
+											{
+												$db->setQuery($query);
+
+												if (!$db->execute(true))
+												{
+													JLog::add(JText::sprintf('JLIB_INSTALLER_ERROR_SQL_ERROR', $db->stderr(true)), JLog::WARNING, 'jerror');
+
+													return false;
+												}
+											}
+										}
+									}
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+
+		return true;
+	}
+
+	/**
+	 * Method to process PHP update files defined in the manifest file
+	 *
+	 * @param   object  $parent  Class calling this method
+	 *
+	 * @return  boolean          True on success
+	 */
+	public function phpUpdates($parent)
+	{
+		$manifest  = $parent->get('manifest');
+
+		if (isset($manifest->update))
+		{
+			if (isset($manifest->update->php) && isset($manifest->update->php->path))
+			{
+				$updatePath = (string) $manifest->update->php->path;
+
+				if ($updatePath != '')
+				{
+					$sourcePath = JPATH_ADMINISTRATOR . '/components/' . $this->extensionName;
+					$db = JFactory::getDbo();
+
+					$files = str_replace('.php', '', JFolder::files($sourcePath . '/' . $updatePath, '\.php$'));
+					usort($files, 'version_compare');
+
+					if (!count($files))
+					{
+						return false;
+					}
+
+					foreach ($files as $file)
+					{
+						if (version_compare($file, $this->oldVersion) > 0)
+						{
+							$this->processPHPUpdateFile($sourcePath . '/' . $updatePath . '/' . $file . '.php', $file);
```

Comment:
you are getting true or false, but you are not actually using those values to break the operation if some of the update files returned false?